backend/src/v2/component/launcher_v2.go

```diff
@@ -696,3 +737,27 @@ func prepareOutputFolders(executorInput *pipelinespec.ExecutorInput) error {
 
 	return nil
 }
+
+// Adds default parameter values if there is no user provided value
+func addDefaultParams(
+	executorInput *pipelinespec.ExecutorInput,
+	component *pipelinespec.ComponentSpec,
+) (*pipelinespec.ExecutorInput, error) {
+	// Make a deep copy so we don't alter the original data
+	executorInputWithDefaultMsg := proto.Clone(executorInput)
+	executorInputWithDefault, ok := executorInputWithDefaultMsg.(*pipelinespec.ExecutorInput)
+	if executorInputWithDefault.GetInputs().GetParameterValues() == nil {
+		executorInputWithDefault.Inputs.ParameterValues = make(map[string]*structpb.Value)
+	}
+	if !ok {
```

Comment:
This check should follow right after when ok got the value.

Reply:
Thank youfor catching this! Added the initialization to the wrong line. I will fix this.